pkg/terraform/terraform_test.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package terraform
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/abcxyz/pkg/testutil"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestTerraform_FormatOutputForGitHubDiff(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name    string
+		content string
+		exp     string
+	}{
+		{
+			name: "replaces_tilde",
+			content: `
+first section -
+first section +
+first section ~
+first section !
+
+    second section -
+    second section +
+    second section ~
+    second section !
+	
+- third section
++ third section
+~ third section
+! third section
+	
+    - fourth section
+    + fourth section
+    -/+ fourth section
+    +/- fourth section
+    ~ fourth section
+    ! fourth section`,
+			exp: `
+first section -
+first section +
+first section ~
+first section !
+
+    second section -
+    second section +
+    second section ~
+    second section !
+	
+- third section
++ third section
+! third section
+! third section
+	
+-     fourth section
++     fourth section
+-/+     fourth section
++/-     fourth section
+!     fourth section
+!     fourth section`,
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			output := FormatOutputForGitHubDiff(tc.content)
+			if got, want := strings.TrimSpace(output), strings.TrimSpace(tc.exp); got != want {
+				t.Errorf("expected\n\n%s\n\nto be\n\n%s\n\n", got, want)
+			}
+		})
+	}
+}
+
+func TestTerraform_GetEntrypointDirectories(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name string
+		dir  string
+		exp  []string
+		err  string
+	}{
+		{
+			name: "success_has_backend",
+			dir:  "../../terraform", // depend on test data in [REPO_ROOT]/terraform
+			exp:  []string{"../../terraform", "../../terraform/has-backend"},
+		},
+		{
+			name: "success_missing_directory",
+			dir:  "../../terraform/missing", // depend on test data in [REPO_ROOT]/terraform
+			exp:  nil,
+			err:  "no such file or directory",
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			tfclient := NewTerraformClient()
+			dirs, err := tfclient.GetEntrypointDirectories(tc.dir)
+			if diff := testutil.DiffErrString(err, tc.err); diff != "" {
+				t.Errorf("unexpected error: %v", diff)
```

Comment:
You can just drop the return value from DiffErrString into t.Error without wrapping it. `t.Error(diff)`